cmd/operator-sdk/add/crd.go

```diff
@@ -102,7 +101,7 @@ func crdFunc(cmd *cobra.Command, args []string) error {
 
 	// This command does not consider an APIs dir. Instead it adds a plain CRD
 	// for the provided resource. We can use NewCRDNonGo to get this behavior.
-	gcfg := gen.Config{}
+	gcfg := gencrd.GeneratorConfig{}
```

Comment:
Perhaps a better pattern would be:
```go
opts := gencrd.GeneratorOptions{
	OperatorName: projectName,
	Resource:     resource,
	CRDVersion:   crdVersion,
	...
}
if err := opts.Generate(); err != nil {
	log.Fatalf("Error generating CRD for %s: %w", resource, err)
}
```

Reply:
This would effectively just equate to making the internal `crdGenerator` struct public rather than having a separate options struct I think, which sounds fine to me unless there was a reason to have them private in the first place?

Reply:
I did it for CRDs in the last commit, is that what you were thinking?

Reply:
Yeah looks great.